drivers/platform/stm32/stm32_gpio_irq.c

```diff
@@ -83,28 +83,50 @@ static bool initialized[STM32_IRQ_CTRL_NB] =  {false};
 /************************ Functions Definitions *******************************/
 /******************************************************************************/
 /**
- * @brief EXTI GPIO Interrupt handler callback
+ * @brief Generic Interrupt handler callback
  * @param pin pin number on which the interrupt occurred (GPIO_PIN_pin)
  */
-void HAL_GPIO_EXTI_Callback(uint16_t pin)
+static void stm32_handle_generic_callback(uint16_t pin)
```

Comment:
I would also declare it `inline` so that there is 1 less entry in the execution call stack.